### PR TITLE
fix cpu/atmega header include guardes.

### PR DESCRIPTION
**Pull request by zhuoshuguo**
Pull request created as per the issue  [#2623](https://github.com/RIOT-OS/RIOT/issues/2623#).

### CODE REVIEW

File: cpu/atmega_common/avr-libc-extra/errno.h [hunk 1]
```diff
@@ -29,8 +29,8 @@
   POSSIBILITY OF SUCH DAMAGE. */
 
 
-#ifndef __ERRNO_H_
-#define __ERRNO_H_ 1
+#ifndef ATMEGA_ERRNO_H_
+#define ATMEGA_ERRNO_H_ 1
```
Comment (zhuoshuguo) [inline cpu/atmega_common/avr-libc-extra/errno.h]: Not pretty sure here..

File: cpu/atmega1281/include/cpu_conf.h [hunk 2]
```diff
@@ -16,8 +16,8 @@
  * @author          Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
  */
 
-#ifndef __CPU_CONF_H
-#define __CPU_CONF_H
+#ifndef ATMEGA_CPU_CONF_H
```
Comment (kaspar030) [inline cpu/atmega1281/include/cpu_conf.h]: Why not CPU_CONF_H?

File: cpu/atmega_common/avr-libc-extra/errno.h (hunk 1 above)
Comment (kaspar030) [inline cpu/atmega_common/avr-libc-extra/errno.h]: same, why not just ERRNO_H?

File: cpu/atmega1281/include/cpu_conf.h (hunk 2 above)
Comment (zhuoshuguo) [inline cpu/atmega1281/include/cpu_conf.h]: I made this kind of revision according to the [comments](https://github.com/RIOT-OS/RIOT/pull/5571) by gebart.

Comment (kaspar030) [inline cpu/atmega1281/include/cpu_conf.h]: > I made this kind of revision according to the [comments](https://github.com/RIOT-OS/RIOT/pull/5571) by gebart.

I think that was meant only for the stellaris headers. RIOT's convention
so far has been FILE_NAME.H. That might be up for debate, though...

Comment (gebart) [inline cpu/atmega1281/include/cpu_conf.h]: yes, that comment was for the Stellaris headers specifically.
In general, you should name the include guard to what the `#include` line will contain.
if you use `#include "mymodule.h"` to include the file, then its include guard will be `MYMODULE_H_`. If you use `#include "mymodule/config.h"` then you should use an include guard name something like `MYMODULE_CONFIG_H_`

The trailing underscore is up for debate though.

**Comment by BytesGalore** [comment]
to give some numbers for a trailing underscore decision,
we have 2213 guards without trailing underscore
``` sh
RIOT$ egrep -r '*._H\>' . | wc
   2213    6919  177051
```
and 1854 with trailing underscores
``` sh
RIOT$ egrep -r '*._H_\>' . | wc
   1854    5453  133055
```
so there's a slight tendency towards no trailing underscores.

**Comment by zhuoshuguo** [comment]
Updated.

File: cpu/atmega1281/include/cpu_conf.h (hunk 2 above)
Comment (zhuoshuguo) [inline cpu/atmega1281/include/cpu_conf.h]: Thanks for the explain. :-)

**Comment by gebart** [comment]
ACK

**Pull request merged**